common/Resources/ti.internal/extensions/node/fs.js

```diff
@@ -0,0 +1,1466 @@
+fs.lstat = (path, options, callback) => fs.stat(path, options, callback);
+fs.lstatSync = (path, options) => fs.statSync(path, options);
+
+/**
+ * @param {string|Buffer|URL} path file path
+ * @param {string|object} [options] options
+ * @param {boolean} [options.recursive=false] recursivley create dirs?
+ * @param {integer} [options.mode=0o777] permissions
+ * @param {errorCallback} callback async callback
+ */
+fs.mkdir = (path, options, callback) => {
+	setTimeout(() => {
+		if (typeof options === 'function') {
+			callback = options;
+			options = { recursive: false, mode: 0o777 };
+		}
+		try {
+			fs.mkdirSync(path, options);
+		} catch (e) {
+			callback(e);
+			return;
+		}
+		callback(null);
+	}, 1);
+};
+
+/**
+ * @param {string|Buffer|URL} path file path
+ * @param {string|object} [options] options
+ * @param {boolean} [options.recursive=false] recursivley create dirs?
+ * @param {integer} [options.mode=0o777] permissions
+ */
+fs.mkdirSync = (path, options) => {
+	const tiFile = getTiFileFromPathLikeValue(path);
+	if (typeof options === 'number') {
+		options = { recursive: false, mode: options };
+	} else {
+		options = mergeDefaultOptions(options, { recursive: false, mode: 0o777 });
+	}
+	if (!tiFile.createDirectory(options.recursive) && !options.recursive) {
+		if (tiFile.exists()) {
+			// already existed!
+			throw fileAlreadyExists('mkdir', path);
+		}
+		// We failed, probably because we didn't ask for recursive and parent doesn't exist, so reproduce node's error
+		throw noSuchFile('mkdir', path);
+	}
+};
+
+/**
+ * @callback tempDirCallback
+ * @param {Error} err - Error if one occurred
+ * @param {string} folder - generated folder name
+ */
+
+/**
+ * @param {string} prefix directory name prefix
+ * @param {string|object} [options] options
+ * @param {string} [options.encoding='utf-8'] prefix encoding
+ * @param {tempDirCallback} callback async callback
+ */
+fs.mkdtemp = (prefix, options, callback) => {
+	assertArgumentType(prefix, 'prefix', 'string');
+	if (typeof options === 'function') {
+		callback = options;
+		options = {};
+	}
+	options = mergeDefaultOptions(options, { encoding: 'utf-8' });
+
+	// try to be all async
+	const tryMkdtemp = () => {
+		// FIXME: use the encoding specified!
+		const generated = randomCharacters(6); // generate six random characters
+		const path = `${prefix}${generated}`;
+		fs.mkdir(path, 0o700, err => {
+			if (err) {
+				if (err.code === 'EEXIST') {
+					// retry!
+					setTimeout(tryMkdtemp, 1);
+					return;
+				}
+				// bubble up error
+				callback(err);
+				return;
+			}
+			// succeeded! Hurray!
+			callback(null, path);
+		});
+	};
+	setTimeout(tryMkdtemp, 1);
+};
+
+/**
+ * Creates a unique temporary directory.
+ * @param {string} prefix directory name prefix
+ * @param {string|object} [options] options
+ * @param {string} [options.encoding='utf-8'] prefix encoding
+ * @returns {string} path to created directory
+ */
+fs.mkdtempSync = (prefix, options) => {
```

Comment:
<!--
  0 failure: 
  1 warning:  common/Resources/...
  
  
  DangerID: danger-id-Danger;
  File: common/Resources/ti.internal/extensions/node/fs.js;
  Line: 518;
-->

- :warning: common/Resources/ti.internal/extensions/node/fs.js line 518 – 'options' is assigned a value but never used. (no-unused-vars)